core/capabilities/triggers/logevent/service.go

```diff
@@ -0,0 +1,157 @@
+}
+
+func (config Config) Version(capabilityVersion string) string {
+	return fmt.Sprintf(capabilityVersion, config.Network, config.ChainID)
+}
+
+type Params struct {
+	Logger         logger.Logger
+	Relayer        core.Relayer
+	RelayerSet     core.RelayerSet
+	LogEventConfig Config
+}
+
+var _ capabilities.TriggerCapability = (*TriggerService)(nil)
+var _ services.Service = &TriggerService{}
+
+// Creates a new Cron Trigger Service.
+// Scheduling will commence on calling .Start()
+func NewLogEventTriggerService(p Params) *TriggerService {
+	l := logger.Named(p.Logger, "LogEventTriggerCapabilityService: ")
+
+	logEventStore := NewCapabilitiesStore[logEventTrigger, capabilities.TriggerResponse]()
+
+	s := &TriggerService{
+		lggr:           l,
+		triggers:       logEventStore,
+		relayer:        p.Relayer,
+		logEventConfig: p.LogEventConfig,
+	}
+	s.CapabilityInfo, _ = s.Info(context.Background())
+	s.Validator = capabilities.NewValidator[RequestConfig, Input, capabilities.TriggerResponse](capabilities.ValidatorArgs{Info: s.CapabilityInfo})
+	return s
+}
+
+func (s *TriggerService) Info(ctx context.Context) (capabilities.CapabilityInfo, error) {
+	return capabilities.NewCapabilityInfo(
+		s.logEventConfig.Version(ID),
+		capabilities.CapabilityTypeTrigger,
+		"A trigger that listens for specific contract log events and starts a workflow run.",
+	)
+}
+
+// Register a new trigger
+// Can register triggers before the service is actively scheduling
+func (s *TriggerService) RegisterTrigger(ctx context.Context, req capabilities.TriggerRegistrationRequest) (<-chan capabilities.TriggerResponse, error) {
+	if req.Config == nil {
+		return nil, errors.New("config is required to register a log event trigger")
+	}
+	reqConfig, err := s.ValidateConfig(req.Config)
+	if err != nil {
+		return nil, err
+	}
+	// Add log event trigger with Contract details to CapabilitiesStore
+	respCh, err := s.triggers.InsertIfNotExists(req.TriggerID, func() (*logEventTrigger, chan capabilities.TriggerResponse, error) {
+		return newLogEventTrigger(ctx, reqConfig, s.logEventConfig, s.relayer)
+	})
+	if err != nil {
+		return nil, fmt.Errorf("LogEventTrigger %v", err)
+	}
+	s.lggr.Infow("RegisterTrigger", "triggerId", req.TriggerID, "WorkflowExecutionID", req.Metadata.WorkflowExecutionID)
+	return respCh, nil
+}
+
+func (s *TriggerService) UnregisterTrigger(ctx context.Context, req capabilities.TriggerRegistrationRequest) error {
+	trigger, ok := s.triggers.Read(req.TriggerID)
+	if !ok {
+		return fmt.Errorf("triggerId %s not found", req.TriggerID)
+	}
+	// Close callback channel and stop log event trigger listener
+	trigger.Stop()
+	// Remove from triggers context
+	s.triggers.Delete(req.TriggerID)
+	s.lggr.Infow("UnregisterTrigger", "triggerId", req.TriggerID, "WorkflowExecutionID", req.Metadata.WorkflowExecutionID)
+	return nil
+}
+
+// Start the service.
+func (s *TriggerService) Start(ctx context.Context) error {
+	if s.relayer == nil {
```

Comment:
Hmmmm... do we need this? We should be able to rely on it being non-nil way before this

Reply:
Followed cron trigger capability as an example here..
We could remove this once we understand what functionality needs to go in Start function